### ISSUE TITLE
Routing error: uninitialized constant Api::V1::Category

### ISSUE THREAD

**Issue by lifeBalance** [hiromipaw/wikicat#5]
Hi, I'm in **page 94** following along, and running the first `curl` throws the error mentioned in the title.
This is my [wikicat project](https://github.com/lifeBalance/wikicat) in case you wanted to check.
Rails **4.2.4** 
Ruby **2.2.2p95**

**Comment by hiromipaw** [comment]
Hey,
did you defined new routes without restarting the server?

**Comment by lifeBalance** [comment]
I wouldn't open an issue without having put a fair amount of fight )) But yeah, I restarted the router like probably a dozen times, and went over the code (it wasn't much, really) quite a few times.

**Comment by hiromipaw** [comment]
Could you use the following syntax in your routes? 

```
  get '/category/:category', :to => 'category#show'
```

Also please respect the spaces. I know it doesn't matter but it makes reading the code easier and everything more consistent for larger projects.
Then you will have the respond_to error. This is because of the updates to Rails and Rails-api. If you downgrade the gems version as in my original repo you will be fine.

Thanks for reaching out!!

Silvia

**Comment by lifeBalance** [comment]
Thanks Silvia, but that syntax is allowed in hashes with symbol keys since version **1.9.**. Also, in your original repo, none of the gems in your `Gemfile` (except `sdoc`) include a version, only Rails does.
One of the things I've noticed in your repo is that your `ApplicationController` inherits from `ActionController::Base` whereas mine inherits from `ActionController::API`. Also, you have a `views/layout` folder with `application.html.erb` layout in it. It looks like you have generated your app with `rails new` instead of `rails-api new`.

**Comment by hiromipaw** [comment]
Hey there!
Yes the syntax is allowed, but something in the way you write it gets it messed. I cloned your repository and did the change suggested and it works. I will send you a PR so you can try it out.
Also there are a few little differences between rails and rails-api, but these shouldn't impact at all the way you build the API at this stage.
To check the version I used in my repository check the Gemfile.lock file.

**Comment by lifeBalance** [comment]
So you say my project works on your machine with the edit on routes? I still get the same error. What's your ruby version?

**Comment by hiromipaw** [comment]
$ ruby -v
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin14]

**Comment by lifeBalance** [comment]
Oh boy, I deleted the project folder, cloned my repo, bundle update(This time I change Rails in the Gemfile to the version you use), and now am getting `NoMethodError
undefined method`raise_in_transactional_callbacks=' for #Class:0x007fba2b575f98`
Does that make any sense?

**Comment by hiromipaw** [comment]
I sent you a PR regarding the routes. Also please do not run bundle update, use the version from my repo by looking at the Gemfile.lock.

**Comment by lifeBalance** [comment]
Yeah, I merged your PR. Now, I've copy/paste your Gemfile.lock, run bundle, and same weird  `NoMethodError undefined methodraise_in_transactional_callbacks`.
Do you think does have anything to do with dropping the db? Before, when I started with your book, I tried the dumps, before I realize one of them was > 10 gb :pensive: , and had cut it out in the middle when I was running out of space in my HDD and DROP the db.
Then I took the seed file solution, and seeded it, of course.

**Comment by hiromipaw** [comment]
It is better to use the information in ghe Gemfile.lock to specify a specific version of a gem in your Gemfile. 
If you have dropped the db, you have to run rake db:setup.
Also I have sent a second PR, please check it.

**Comment by lifeBalance** [comment]
Wow, girl you are tough, I don't know where are you based but here in Moscow(I'm spanish though) is past 5, so I'm gonna wrap up. Thanks for your help.

**Comment by lifeBalance** [comment]
Hey Silvia, I got it working. It was indeed a silly eye-straining error on my side, that slipped our defenses:

``` ruby
 get 'category/:category', :to => 'category/#show'
```

See that slash before the `#show` action? :scream_cat: 

Anyways, I ended up starting over here [wicked](https://github.com/lifeBalance/wicked) where I used the most up to date versions of all gems, hopefully I won't find problems along the line.

Indeed I hit the `respond_to` error you mentioned, because from **Rails 4.2.***, class-level methods such as `respond_to` have been moved to the [responders gem](https://github.com/plataformatec/responders), so instead of reverting to an older Rails version, I just added it to my `Gemfile`.

**Comment by bifork** [comment]
I hit errors when I reach this line of your book, too.
Rails 5 and latest version of ActiveModelSerializers.

I wish you could rewrite the book with more business-model-centered examples. The topics covered by your book is REALLY good. But the examples don't work under the current versions of gems.

Really, there are no comparable good books introducing how to develop a production-level RESTful API in Rails. The old Rails books don't introduce API-only apps while the market these days demand such skills.

These days, web apps have backend totally separated from frontend. But the tutorials of the Rails community seems to be out-of-date.

If you can write a book introducing the whole process, featuring a business model like simplified versions of AirBnb or Twitter or Instagram or YouTube, that would be the best-selling and prime tutorial for many would-be startups.

It would be ideal if you examples include authentication, both the email confirmation process or registration with google or twitter accounts.

If you can integrate your backend example with React or Angular2 frontend, it would be even greater.

Thank you. Your book really helped me anyway.

**Comment by bifork** [comment]
While there are not good books about building APIs in Rails, competing technologies are taking over market share. I really cannot find a better books of yours. If you can pick up the banner and produce a good book, you can be the best tutor and well-known programmer in the world. If you can help new programmers to get started with the high-efficiency Rails framework to the demand of modern market, you would be truly awesome and famous.

BTW, introduce fewer abstract concepts in your new book. Try to present critical example codes. Get the real app done. Let readers read fast. You can put the concepts into URLs appended to each chapter.

**Comment by hiromipaw** [comment]
Hey @bifork, 
Thanks for your comments, some of the errors are related to old version of Rails. I am sorry about those, but I wrote the book 2 years ago now. I am planning on releasing an update, hopefully in the spring.

**Comment by ghost** [comment]
Thank @hiromipaw for your reply, Silvia.

Though I cannot hope for a big revamp in your spring release. But I do wish you could take time to expand your wonderful topics to greater detail. Right now, your book seems to be friendly to really experienced developers.

I wish you could include the user model and authentication solution from the start. Then a RESTful resources owned by a user and conversations/comments about a RESTful resource object between users. Besides the relationship/association between models, if you can include a transaction like payment or ownership transfer, it would be all too great.

I can imagine you are incredibly busy, especially for a girl who pursues highest academic degree besides cutting-edge programming skills. But I believe, if you spend time to write an awesome popular book, you can make money while you're sleeping or playing on your beautiful beaches.

I wish I could be a great author after becoming very skilled. But I'm not experienced yet and still slowly researching. A foggy world to me right now.

Muchas gracias:)
